lib/ecto/association.ex

```diff
@@ -9,8 +9,19 @@ defmodule Ecto.Association.NotLoaded do
     * `__field__` - the association field in `owner`
     * `__owner__` - the schema that owns the association
     * `__cardinality__` - the cardinality of the association
+  """
 
+  @typedoc """
+  * `__field__` - the association field in `owner`
+  * `__owner__` - the schema that owns the association
+  * `__cardinality__` - the cardinality of the association
   """
+  @type t :: %__MODULE__{
+    __field__: atom(),
```

Comment:
because of underscores it seems to me it's private-ish, maybe the the `t()` type should be opaque or these fields private?

Reply:
They are not private. The reason why they are underscored is to avoid accidental matches in code like this `comment.post.owner` for the case the post is not loaded. :)

Reply:
nevermind, the fields are documented so they're public. the reason it's `__field` was probably to minimize the chance of having the same fields in this struct as well as loaded schema struct. Sorry for noise, I'm gonna remove both comments.